### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by kaiyazeera** [The-Cataclysm-Preservation-Project/TrinityCore#359]
**Description:**

The build process exists with error 2.

**Current behaviour:**

```
$ make -j $(nproc) install
Consolidate compiler generated dependencies of target Detour
[  0%] Built target Detour
Consolidate compiler generated dependencies of target Recast
Consolidate compiler generated dependencies of target fmt
[  0%] Built target fmt
[  1%] Built target Recast
Consolidate compiler generated dependencies of target sfmt
[  1%] Built target sfmt
Consolidate compiler generated dependencies of target gsoap
[  1%] Built target gsoap
Consolidate compiler generated dependencies of target jemalloc
Consolidate compiler generated dependencies of target g3dlib
[  3%] Built target jemalloc
[  6%] Built target g3dlib
[  6%] Built target revision_data.h
Consolidate compiler generated dependencies of target libzmq
Consolidate compiler generated dependencies of target common
[  9%] Built target common
Consolidate compiler generated dependencies of target connection_patcher
[ 14%] Built target libzmq
Consolidate compiler generated dependencies of target database
Consolidate compiler generated dependencies of target vmap4assembler
Consolidate compiler generated dependencies of target storm
[ 14%] Built target connection_patcher
[ 14%] Built target vmap4assembler
Consolidate compiler generated dependencies of target zmqpp
[ 16%] Built target database
[ 17%] Built target zmqpp
Consolidate compiler generated dependencies of target shared
[ 18%] Built target shared
Consolidate compiler generated dependencies of target ipc
[ 31%] Built target storm
[ 32%] Built target ipc
Consolidate compiler generated dependencies of target mapextractor
Consolidate compiler generated dependencies of target vmap4extractor
Consolidate compiler generated dependencies of target mmaps_generator
[ 33%] Built target mapextractor
[ 34%] Built target vmap4extractor
[ 34%] Built target mmaps_generator
Consolidate compiler generated dependencies of target bnetserver
[ 34%] Linking CXX executable bnetserver
Consolidate compiler generated dependencies of target scripts
Consolidate compiler generated dependencies of target game
[ 52%] Built target game
[ 99%] Built target scripts
[ 99%] Building CXX object src/server/worldserver/CMakeFiles/worldserver.dir/cmake_pch.hxx.gch
/usr/bin/ld: ../database/libdatabase.a(Transaction.cpp.o): in function `TransactionTask::Execute()':
/home/wow-server/Cata/TrinityCore/src/server/database/Database/Transaction.cpp:80: undefined reference to `std::basic_ostream<char, std::char_traits<char> >& std::operator<< <char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, std::thread::id)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/server/bnetserver/CMakeFiles/bnetserver.dir/build.make:444: src/server/bnetserver/bnetserver] Error 1
make[1]: *** [CMakeFiles/Makefile2:1237: src/server/bnetserver/CMakeFiles/bnetserver.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 99%] Building CXX object src/server/worldserver/CMakeFiles/worldserver.dir/CommandLine/CliRunnable.cpp.o                                                                                                        
[ 99%] Building CXX object src/server/worldserver/CMakeFiles/worldserver.dir/TCSoap/TCSoap.cpp.o
[ 99%] Building CXX object src/server/worldserver/CMakeFiles/worldserver.dir/Main.cpp.o
[ 99%] Building CXX object src/server/worldserver/CMakeFiles/worldserver.dir/RemoteAccess/RASession.cpp.o
In file included from /home/wow-server/Cata/TrinityCore/src/server/shared/DataStores/DB2Store.h:22,
                 from /home/wow-server/Cata/TrinityCore/src/server/game/DataStores/DB2Stores.h:21,
                 from /home/wow-server/Cata/TrinityCore/src/server/game/Globals/ObjectMgr.h:26,
                 from /home/wow-server/Cata/TrinityCore/src/server/worldserver/CommandLine/CliRunnable.cpp:23:
/home/wow-server/Cata/TrinityCore/src/server/shared/DataStores/DBStorageIterator.h:25:39: warning: ‘template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator’ is deprecated [-Wdeprecated-declarations]
   25 | class DBStorageIterator : public std::iterator<std::forward_iterator_tag, T>
      |                                       ^~~~~~~~
In file included from /usr/include/c++/12.1.0/bits/stl_algobase.h:65,
                 from /usr/include/c++/12.1.0/memory:63,
                 from /home/wow-server/Cata/TrinityCore/src/common/Common.h:22,
                 from /home/wow-server/Cata/TrinityCore/src/server/worldserver/PrecompiledHeaders/worldPCH.h:18,
                 from /home/wow-server/Cata/TrinityCore/build/src/server/worldserver/CMakeFiles/worldserver.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/c++/12.1.0/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
/home/wow-server/Cata/TrinityCore/src/server/worldserver/RemoteAccess/RASession.cpp: In member function ‘void RASession::Start()’:
/home/wow-server/Cata/TrinityCore/src/server/worldserver/RemoteAccess/RASession.cpp:35:27: error: ‘sleep_for’ is not a member of ‘std::this_thread’
   35 |         std::this_thread::sleep_for(std::chrono::milliseconds(100));
      |                           ^~~~~~~~~
make[2]: *** [src/server/worldserver/CMakeFiles/worldserver.dir/build.make:125: src/server/worldserver/CMakeFiles/worldserver.dir/RemoteAccess/RASession.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1345: src/server/worldserver/CMakeFiles/worldserver.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Operating system:** Arch Linux

**Comment by kaiyazeera** [comment]
Oddly, when I retried today, the first error occurred here:

```
[ 37%] Linking CXX executable bnetserver
/usr/bin/ld: ../database/libdatabase.a(Transaction.cpp.o): in function `TransactionTask::Execute()':
/home/wow-server/Cata/TrinityCore/src/server/database/Database/Transaction.cpp:80: undefined reference to `std::basic_ostream<char, std::char_traits<char> >& std::operator<< <char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, std::thread::id)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/server/bnetserver/CMakeFiles/bnetserver.dir/build.make:444: src/server/bnetserver/bnetserver] Error 1
make[1]: *** [CMakeFiles/Makefile2:1237: src/server/bnetserver/CMakeFiles/bnetserver.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by kaiyazeera** [comment]
I am not looking for the perfect solution, but if someone could give me a hint how to understand the error and what I could try out, I might contribute a solution.

**Comment by pangolp** [comment]
Try the PR I created a week ago.
https://github.com/The-Cataclysm-Preservation-Project/TrinityCore/pull/360
With those changes, you should be able to compile.

**Comment by kaiyazeera** [comment]
Thank you very much, I will try and report back.

**Comment by kaiyazeera** [comment]
Kudos, problem solved with #360
